spatial_cells/velocity_block_container.h

```diff
@@ -749,18 +749,18 @@ namespace vmesh {
    inline ARCH_HOSTDEV size_t VelocityBlockContainer::sizeInBytes() const {
 #ifdef USE_GPU
       #if defined(__CUDA_ARCH__) || defined(__HIP_DEVICE_COMPILE__)
-      return block_data.size()*WID3*sizeof(Realf) + parameters.size()*sizeof(Real);
+      return block_data.size()*sizeof(Realf) + parameters.size()*sizeof(Real);
       #else
       #ifdef DEBUG_VBC
       const size_t currentSize = block_data.size() / WID3;
       if (currentSize != cachedSize) {
          printf("VBC CHECK ERROR: cached size mismatch, %lu vs %lu in %s : %d\n",currentSize,cachedSize,__FILE__,__LINE__);
       }
       #endif
-      return cachedSize*WID3*sizeof(Realf) + cachedSize*BlockParams::N_VELOCITY_BLOCK_PARAMS*sizeof(Real);
```

Comment:
Commented on this on the actual commit but that might have got lost - on this line WID3 is needed. 

`vector.size()` returns the actual count of elements inside the vector, whereas `cachedSize` is how many blocks are in there - each of size `WID3`.